infra/token-transfer-server/src/lib/transfer.js

```diff
@@ -169,21 +168,20 @@ async function confirmTransfer(transfer, user) {
 /**
  * Sends a blockchain transaction to transfer tokens and waits for the transaction to get confirmed.
  * @param {Transfer} transfer: DB model Transfer object
- * @param {{tokenMock:Object, networkId:number }} opts: options
  * @returns {Promise<{txHash: string, txStatus: string}>}
  */
-async function executeTransfer(transfer, opts) {
-  const { networkId, tokenMock } = opts
-
+async function executeTransfer(transfer) {
   const user = await hasBalance(transfer.userId, transfer.amount, transfer)
 
-  // Setup token library. tokenMock is used for testing.
-  const token = tokenMock || new Token(networkId)
+  await transfer.update({ status: enums.TransferStatuses.Processing })
 
+  // Setup token library
+  const token = new Token(networkId)
   // Send transaction to transfer the tokens and record txHash in the DB.
   const naturalAmount = token.toNaturalUnit(transfer.amount)
   const supplier = await token.defaultAccount()
   const txHash = await token.credit(transfer.toAddress, naturalAmount)
```

Comment:
token.credit may throw an exception for example if the hotwallet does not have enough funds.
should we wrap this call in a try/catch statement and set the transfer status to failed?

Reply:
Done!